example/lib/main.dart

```diff
@@ -377,18 +315,25 @@ class _MyAppState extends State<MyApp> {
       print('Sendimg rest message...');
       dynamic data = messagesToPublish[(realtimePublishCounter++ % messagesToPublish.length)];
       ably.Message m = ably.Message()..data=data..name='Hello';
-      if(typeCounter%3 == 0){
-        await _realtime.channels.get('test-channel').publish(name: 'Hello', data: data);
-      } else if (typeCounter%3 == 1) {
-        await _realtime.channels.get('test-channel').publish(message: m);
-      } else if (typeCounter%3 == 2) {
-        await _realtime.channels.get('test-channel').publish(messages: [m, m]);
-      }
-      if(realtimePublishCounter!=0 && realtimePublishCounter % messagesToPublish.length == 0){
-        typeCounter++;
+      try {
+        if (typeCounter % 3 == 0) {
+          await _realtime.channels.get('test-channel').publish(
+            name: 'Hello', data: data);
+        } else if (typeCounter % 3 == 1) {
+          await _realtime.channels.get('test-channel').publish(message: m);
+        } else if (typeCounter % 3 == 2) {
+          await _realtime.channels.get('test-channel').publish(
+            messages: [m, m]);
+        }
+        if (realtimePublishCounter != 0 &&
```

Comment:
Just a suggestion to not repeat `typeCounter % 3 ==` multiple times
```dart
                  switch (typeCounter % 3) {
                    case 0:
                      await _realtime.channels
                          .get('test-channel')
                          .publish(name: 'Hello', data: data);
                      break;
                    case 1:
                      await _realtime.channels
                          .get('test-channel')
                          .publish(message: m);
                      break;
                    case 2:
                      await _realtime.channels
                          .get('test-channel')
                          .publish(messages: [m, m]);
                      break;
                  }
```

Reply:
Thanks! addressed 91c3389641f516dfc377e2499dc760c871ae5798